installation and os monitoring changes

ua0088 opened the pull request:
installation and os monitoring changes

ua0088 committed:
installation and os monitoring changes

zev333 commented on admin/installation.rst:
```diff
@@ -1200,6 +1200,36 @@ in the same directory as nxmc.properties, correct entry will be:
+java -Dnxmc.logfile=/opt/netxms-webui/logs/nxmc.log -jar /opt/jetty-home-12/start.jar
+
+9. Configure systemctl, run below and edit as per requirents
+
```
I would add systemctl example here

zev333 commented on admin/installation.rst:
```diff
@@ -1200,6 +1200,36 @@ in the same directory as nxmc.properties, correct entry will be:
+7. Generate ssl key
+
+keytool -genkeypair -alias jetty -keyalg RSA -keysize 2048 -keystore /opt/netxms-webui/etc/keystore.p12 -storetype PKCS12 -storepass password -keypass password -validity 3650 -dname "CN=netxms-webui, OU=netxms, O=netxms, L=netxms, ST=netxms, C=netxms"
+sed 's,# jetty.sslContext.keyStorePassword=,jetty.sslContext.keyStorePassword=password,' -i'' start.d/ssl.ini
```
I would add note here about command line options that should be adjusted by user like name and passowrd

ua0088 committed:
DCI section update

ua0088 commented:
Added how-to for jetty

zev333 commented on admin/concepts.rst:
```diff
@@ -485,7 +485,7 @@ Metrics can be collected from multiple data sources:
        storage this way. 
    * - Modbus
-     -
+     - Data is collected via Modbus-TCP industrial protocol. See Modbus for more information.
```
Add reference to Modbus chapter or wiki depending on where you are trying to send reader

zev333 commented on admin/data-collection.rst:
```diff
@@ -46,13 +45,13 @@ It is also possibility to push data to server. If DCI source is set to
+server and agent is broken as poll request would not get to agent. There is
+special configuration that allows data collection and storage on agent till
+connection with server is restored and collected data is pushed to the server thereafter.
+This option is available for metrics, table metrics and proxy SNMP metrics. Functionality is not
```
https://track.radensolutions.com/issue/NX-860
https://track.radensolutions.com/issue/NX-713

Both issues fixed and work with caching mode

zev333 commented on admin/data-collection.rst:
```diff
@@ -63,11 +62,11 @@ DCI configuration
+data collection tab, click on node object in
+:guilabel:`Infrastructure` or  :guilabel:`Network` perspective, and click
+:guilabel:`Data Collection` tab. You will see the list of configured
+data collection items. From here, you can add new or change existing metrics for
```
Now as DCI config and Last values are combined. We can mention that from there we can see collected data and configure new one or edit existing one. (You can write in your own words)

zev333 commented on admin/data-collection.rst:
```diff
@@ -146,35 +145,14 @@ via :ref:`nxapush-label` or :ref:`nxpush-label` command line tool) instead of
-     - Value is generated by NXSL script. Script should be stored in
-       :guilabel:`Script Library`. Sctipt should return `Table class
-       <https://www.netxms.org/documentation/nxsl-latest/#class-table>`_.
+Possible table metric origins are identical to single-value origins described above.
```
It's not right UI does not reflects real situation.
Possible table sources: Internal, agent, SNMP, Script

zev333 commented on admin/data-collection.rst:
```diff
@@ -146,35 +145,14 @@ via :ref:`nxapush-label` or :ref:`nxpush-label` command line tool) instead of
 Data Type
 ~~~~~~~~~
 
 Data type for the metric. Can be one of the following: :guilabel:`Integer`,
```
I think it's copy paste as data type is for table columns not for the metric itself.

zev333 commented on admin/data-collection.rst:
```diff
@@ -184,14 +162,14 @@ Source node override
 ~~~~~~~~~~~~~~~~~~~~
```
Source node override should be removed from this chapter and moved to the chapter where page with this configuration is described

zev333 replied:
Plese also check other chapters accordingly and add missing for example units

zev333 commented on admin/data-collection.rst:
```diff
@@ -447,23 +420,47 @@ Data collection error    An indicator of data collection error. Instead of
+                         current time of the day on the same day of the week 
+                         for last 10 weeks. If new data point is classified as 
+                         outlier in both data sets, DCI will be marked as having 
+                         anomalous value. 
```
Maybe we should add that it will create additional load to database. 
And also we can add about algorithm: https://www.netxms.org/forum/general-support/dci-detect-anomalies/

zev333 commented on admin/data-collection.rst:
```diff
@@ -824,7 +836,7 @@ Other available options:
 
-This configuration part can be used for free for text comments. To make
+This configuration part can be used freely for text comments in order to make
 additional notes about DCI configuration or usage.
```
It's also added to alarm as a comment when alarm is generated on threshold event so it can be used for solutions description for operator to know how to solve the problem

zev333 commented on admin/data-collection.rst:
```diff
@@ -859,14 +871,14 @@ Table DCIs
 sets that can be acquired together or for atomic collection. Atomic collection 
 is when you need to take a data snapshot that consists of multiple items 
-collected at the exact same time.
+collected at the exact same time. Right-click on non string value to access history and build charts.
```
You can access fistory for any value string and non string line chart is available only for non string values

zev333 commented on admin/data-collection.rst:
```diff
@@ -899,17 +911,17 @@ Agent caching mode
 :ref:`how_data_collection`.
 
-Agent side cache is configurable globally, on node level, and on DCI level. By
+Agent side cache is configurable globally on node and DCI levels. By
```
It's 3 levels to configure cache functionality. You can change this configuration independently on each level.

zev333 commented on admin/data-collection.rst:
```diff
@@ -958,6 +997,10 @@ threshold for CPU utilization. To avoid these problems, you can use data
 collection templates. Data collection template (or just template for short) is
 a special object, which can have configured DCIs similar to nodes.
 
```
Maybe we should add Information that it's separate perspective (before they were in one tree so there was no need to mention that).

zev333 commented on admin/data-collection.rst:
```diff
@@ -981,52 +1024,54 @@ Creating template
-.. figure:: _images/remove_template.png
+.. figure:: _images/dci_remove_template.png
 
-If you select Unbind DCIs from template, all DCIs related to template will
```
Information about template removal from the node is missing in new text. Small explanation about options in remove dialog meaning.

zev333 replied:
Explanation about unbind DCI from template still missing in new text.

zev333 commented on admin/data-collection.rst:
```diff
@@ -1074,6 +1119,6 @@ Working with collected data
 data and work with it in different ways. Data can be visualized in three ways:
 in graphical form, as a historical view(textual format) and as DCI summary table,
-this layout types can be combined in Dashboards.
+this layout types can be combined in Dashboards and Graphs.
```
Graph can't contian graph(graphical form) or summary table or last values as text. Only dashboards can combine other elements

zev333 commented on admin/installation.rst:
```diff
@@ -1200,6 +1200,44 @@ in the same directory as nxmc.properties, correct entry will be:
+systemctl edit --force --full netxms-webui.service
+
+
+.. figure:: _images/systemctl_edit.png
```
I was thinking about text example added as a .. code-block::

zev333 commented on admin/installation.rst:
```diff
@@ -1200,6 +1200,44 @@ in the same directory as nxmc.properties, correct entry will be:
+keytool -genkeypair -alias jetty -keyalg RSA -keysize 2048 -keystore /opt/netxms-webui/etc/keystore.p12 -storetype PKCS12 -storepass password -keypass password -validity 3650 -dname "CN=netxms-webui, OU=netxms, O=netxms, L=netxms, ST=netxms, C=netxms"
+sed 's,# jetty.sslContext.keyStorePassword=,jetty.sslContext.keyStorePassword=password,' -i'' start.d/ssl.ini
+
+Adjust configurable values like path, user and password as per requirements.
```
I would add it as a note or warning to be more visible: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html

zev333 commented on admin/_images/systemctl_edit.png:
Do not see file to be used somewhere. Maybe should be deleted?

zev333 commented on admin/installation.rst:
```diff
@@ -1200,6 +1200,65 @@ in the same directory as nxmc.properties, correct entry will be:
+   [Install]
+    WantedBy=multi-user.target
+    EnableDefaultCounters = yes
+
```
I would add note that jetty user and group should be created or exist in order to use them in service file

zev333 commented on admin/data-collection.rst:
```diff
@@ -15,27 +15,22 @@ Every node can have many data collection items configured (see
+node DCI capability information displayed in :guilabel:`Overview`-> :guilabel:`Capabilities` 
+section as well as last values in :guilabel:`Data Collection` tab view.
+
+Supplimentary DCI representation can be configured in
```
Chapter separation looks odd. 

I would separate chapter about displaying data from the chapter that describes how data collection is done.

Like:

Rework of text about Capabilities. 

Supplimentary DCI representation can be configured in :ref:`dci-othe-options-label` or :ref:`dci-performance-tab`. 

All configured DCIs are checked for polling requirement every second. If DCI needs to be polled, appropriate polling request is placed into internal ...

zev333 commented on admin/data-collection.rst:
```diff
@@ -184,14 +159,14 @@ Source node override
-Data is collected from the same node if no value set.
+Data is collected from the local node if no value is set.
 
 Polling
```
If we will check screenshot Polling is called part with Data type, units and multuplier option. Not part about how often to poll. So this part should be renamed as "Collection schedule" and "Polling" part need to be added movind inside Data type part

zev333 commented on admin/data-collection.rst:
```diff
@@ -306,65 +270,71 @@ Data Transformations
 None                No delta calculation performed. This is the default
                     setting for newly created DCI.
-Simple              Resulting value will be calculated as a difference
+Simple              Resulting value will be calculated as difference
```
Katya said that there should be article "a" in this and next lines

zev333 commented on admin/data-collection.rst:
```diff
@@ -63,15 +58,16 @@ DCI configuration
+:guilabel:`Infrastructure` or  :guilabel:`Network` perspective, and click
+:guilabel:`Data Collection` tab. You will see the list of configured
+data collection items. From here, since DCI configuration and Last values are combined, 
+one can see collected data and configure new or change existing metrics for monitoring. Right click on item 
```
There should be and article "Right click on an item ..."

zev333 commented on admin/data-collection.rst:
```diff
@@ -200,72 +175,61 @@ Polling mode and interval describe schedule type and interval between consecutiv
+every second and displayed on dashboards in close to real time.
 
-    - :guilabel:`Save only changed values` - if enabled, value is saved to the
+    - :guilabel:`Save only changed values` - if enabled, value is saved in
```
in the database
There should be article as it's exact our server database

zev333 commented on admin/data-collection.rst:
```diff
@@ -306,65 +267,71 @@ Data Transformations
-passed to the transformation script as a parameter; and a result of script
-execution is a final DCI value. Transformation script gets original value as
+transformation script is applied, the resulting value of the first step is
+passed to the transformation script as a parameter; and result of script
```
missing article should be replaced as the article

zev333 commented on admin/data-collection.rst:
```diff
@@ -749,7 +740,7 @@ If binary value is returned, it has the following meaning:
 
 If an array is returned, then instance is counted as accepted. Only first element 
-of the array is obligatory, the rest elements are optional (but to include an 
+of the array is compulsory, the rest elements are optional (but to include an 
```
Please replace compulsory to mandatory

zev333 commented on admin/data-collection.rst:
```diff
@@ -762,24 +753,17 @@ NetObj                   Object connected with this :term:`DCI`
+Performance view
+----------------
 
-Multiple DCIs can be grouped in one graph. To group them use the same group name
```
this part is missing form new documentation

zev333 commented on admin/data-collection.rst:
```diff
@@ -824,8 +822,9 @@ Other available options:
-This configuration part can be used for free for text comments. To make
-additional notes about DCI configuration or usage.
+This configuration part can be used freely for text comments in order to make
+additional notes about DCI configuration or usage. Those comments are added to threshold alarm itself, for example, 
```
Please replace:
This configuration part can be used freely for text comments in order to make additional notes about DCI configuration or usage. Those comments are added to threshold alarm itself, for example, and can be used by operator to indicate solution thus reducing impact.


With:
This configuration page can be used freely for text comments to add additional notes about DCI configuration or usage. These comments are added to alarms created from threshold violation events. For example, they can be used to inform operators about problem-solving approaches.

zev333 commented on admin/data-collection.rst:
```diff
@@ -420,54 +387,77 @@ Threshold Configuration
+                         current time of the day on the same day of the week 
+                         for last 10 weeks. If new data point is classified as 
+                         outlier in both data sets, DCI will be marked as having 
+                         anomalous value. Using this setting may adversly affect your database performance. Please size accordigly.
```
Replace:
Please size accordigly.
With:
Note: This is an experimental feature - use with caution.

zev333 commented on admin/data-collection.rst:
```diff
@@ -15,27 +15,19 @@ Every node can have many data collection items configured (see
+
+Node capabilities might give some information about node available sources for data collection in :guilabel:`Overview`-> :guilabel:`Capabilities` section. Please also refer to 
+:guilabel:`Data Collection` tab view for last values from DCI collections. 
+Supplimentary DCI representation can be configured in :ref:`dci-othe-options-label` or :guilabel:`Performance` tab.
```
Please create reference to Performance View chapter.
Let's replace it whit this text:

Node capabilities provide information about available sources for data collection in the :guilabel:Overview -> :guilabel:Capabilities section.

The last values of DCIs for the node can be found on the :guilabel:`Data Collection` tab. Additionally, specific DCIs can be displayed in the :guilabel:`Overview` -> :guilabel:`Last Values` section or as a graph on the :guilabel:`Performance` tab. More details about DCI display configuration options can be found in the :ref:`dci-other-options-label` and :guilabel:`dci-performance-view` chapters.

zev333 commented on admin/data-collection.rst:
```diff
@@ -184,14 +162,14 @@ Source node override
 Source node of metrics collection. This can be used when other node provides
-information about this node. In this way collected data can be collected and
-shown on right nodes.
+information about local node. In this way, platform provides additional flexability of where 
```
Let's write current node not local

zev333 commented on admin/data-collection.rst:
```diff
@@ -762,24 +753,17 @@ NetObj                   Object connected with this :term:`DCI`
 
-Multiple DCIs can be grouped in one graph. To group them use the same group name
-in "Group" field.
+This section provides shortcut configuration option to enable visual representation of 
```
Please replace:
This section provides shortcut configuration option to enable visual representation of collected DCI values in some graphical way. Not available for table metrics. Select :guilabel:`Show in Performace View`, save and DCI historic data will be added in Performance tab as graph in addition to some default graphs, which come with installation. There are different option to tune visual representation. More on visualization can be accessed in designated chapter here :ref:`visualisation`.

With (Also format note section):
This section provides configuration options for displaying DCI values as line charts on the :guilabel:`Performance` tab. Various options are available to visually represent the collected data; see :ref:`visualisation` for more details.

Note: Not available for table metrics.

zev333 commented on admin/data-collection.rst:
```diff
@@ -930,17 +930,44 @@ It can be configured:
 
-Last DCI values View
-====================
+Data Collection tab view
```
Data Collection tab
without view

ua0088 committed:
Merge branch 'netxms:master' into master

zev333 commented on admin/_images/dci_access_contr_page.png:
Not needed as there is admin/_images/dci_access_control_page.png image

zev333 commented on admin/data-collection.rst:
```diff
@@ -794,12 +783,28 @@ is able to change and see DCI configuration.  When list is not empty, then both
 request.
 
-.. figure:: _images/dci_access_control_page.png
+.. figure:: _images/dci_access_contr_page.png
```
Please use dci_access_control_page.png

zev333 commented on admin/_images/dci_tab.png:
Please remove image as it's duplicate of admin/_images/dci_table.png

zev333 commented on admin/data-collection.rst:
```diff
@@ -859,14 +863,15 @@ Table DCIs
 values that can be collected simultaneously. 
 
-.. figure:: _images/dci_table.png
+.. figure:: _images/dci_tab.png
```
Please use dci_table.png

zev333 commented on admin/_images/dci_transf_page.png:
Please remove this one as its duplicate of admin/_images/dci_transformation_page.png

zev333 commented on admin/data-collection.rst:
```diff
@@ -335,36 +299,40 @@ Average per minute  Resulting value will be calculated as a difference
 
 
-.. figure:: _images/dci_transformation_page.png
+.. figure:: _images/dci_transf_page.png
```
Please use dci_transformation_page.png

zev333 commented on admin/data-collection.rst:
```diff
@@ -15,27 +15,20 @@ Every node can have many data collection items configured (see
+
+Node capabilities provide information about available sources for data collection in the :guilabel:`Overview`-> :guilabel:`Capabilities` section. Please also refer to 
+:guilabel:`Data Collection` tab view for last values from DCI collections. 
+The last values of DCIs for the node can be found on the :guilabel:`Data Collection` tab. Additionally, specific DCIs can be displayed in the :guilabel:`Overview`` -> :guilabel:`Last Values section` 
```
Out of this sentences:
Please also refer to :guilabel:`Data Collection` tab view for last values from DCI collections. 
The last values of DCIs for the node can be found on the :guilabel:`Data Collection` tab. 

Please remove first and keep second as they mean the same.

zev333 commented on admin/data-collection.rst:
```diff
@@ -184,88 +157,77 @@ Source node override
 Source node of metrics collection. This can be used when other node provides
-information about this node. In this way collected data can be collected and
-shown on right nodes.
+information about current node. In this way, platform provides additional flexability of where 
```
spelling:
flexibility

zev333 commented on admin/_images/threshold_conf_dia.png:
Please delete  copy of admin/_images/threshold_configuration_dialog.png

zev333 commented on admin/data-collection.rst:
```diff
@@ -420,54 +388,77 @@ Threshold Configuration
 When adding or modifying a threshold, you will see the following dialog:
 
-.. figure:: _images/threshold_configuration_dialog.png
+.. figure:: _images/threshold_conf_dia.png
```
please use threshold_configuration_dialog.png

zev333 commented on admin/data-collection.rst:
```diff
@@ -762,28 +754,25 @@ NetObj                   Object connected with this :term:`DCI`
+  Note: Not available for table metrics.
 
 
-.. figure:: _images/dci_performance_tab_page.png
```
Please rename _images/dci_perf_view_page.png to _images/dci_performance_tab_page.png and use that name here. 

Our standard is not to shorten words if it is possible.

zev333 commented on admin/data-collection.rst:
```diff
@@ -794,12 +783,28 @@ is able to change and see DCI configuration.  When list is not empty, then both
 
     DCI configuration access control property page
 
 .. _dci-othe-options-label:
```
Remove reference it's duplicate

zev333 commented on admin/data-collection.rst:
```diff
@@ -971,62 +1009,65 @@ all such DCIs or leave them, but remove relation to the template. If you delete
 -----------------
 
-To create a template, right-click on :guilabel:`Template Root` or
-:guilabel:`Template Group` object in the :guilabel:`Object Browser`, and click
```
:guilabel:`Template Root` or
:guilabel:`Template Group` 

Please keep them as whey were before as they are internal terms:
https://netxms.org/documentation/adminguide/concepts.html#objects

zev333 commented on admin/data-collection.rst:
```diff
@@ -971,62 +1009,65 @@ all such DCIs or leave them, but remove relation to the template. If you delete
-to unbind from template, and click :guilabel:`OK`. The system will ask you how
-to deal with DCIs configured on node and associated with template:
-
-.. figure:: _images/remove_template.png
```
Please delete pictures that are not used by documentation.

ua0088 committed:
Delete admin/_images/dci_access_contr_page.png

ua0088 committed:
Update data-collection.rst

ua0088 committed:
Update data-collection.rst

ua0088 committed:
Delete admin/_images/dci_tab.png

ua0088 committed:
Update data-collection.rst

ua0088 committed:
Delete admin/_images/dci_transf_page.png

ua0088 committed:
Update data-collection.rst

ua0088 committed:
Update data-collection.rst

spelling flexibility

ua0088 committed:
Update data-collection.rst

added threshold_configuration_dialog.png

ua0088 committed:
Delete admin/_images/threshold_conf_dia.png

ua0088 committed:
Update data-collection.rst

corrected dci_performance_tab_page.png

ua0088 committed:
Delete admin/_images/dci_performance_tab_page.png

ua0088 committed:
added new version of dci_performance_tab_page.png

ua0088 committed:
Delete admin/_images/dci_perf_view_page.png

ua0088 committed:
Update data-collection.rst

ua0088 committed:
Delete admin/_images/remove_template.png

ua0088 committed:
Update data-collection.rst

added 

If you select Unbind DCIs from template, all DCIs related to template will remain configured on a node, but association between the DCIs and template will be removed. Any further changes to the template will not be reflected in these DCIs. If you later reapply the template to the node, you will have two copies of each DCI - one standalone (remaining from unbind operation) and one related to template (from new apply operation). Selecting Remove DCIs from node will remove all DCIs associated with the template. After you click OK, node will be unbound from template.

ua0088 committed:
Update data-collection.rst
